engines/director/lingo/lingo-code.cpp

```diff
@@ -1307,13 +1327,13 @@ Datum LC::compareArrays(Datum (*compareFunc)(Datum, Datum), Datum d1, Datum d2,
 }
 
 Datum LC::eqData(Datum d1, Datum d2) {
-	// Lingo doesn't bother checking list equality if the left is longer
+	// Lingo only checks array equality if the sizes are the same
```

Comment:
Hmmm... My tests were showing otherwise, at least in earlier Directors. Could you please recheck?

Reply:
Can you go into detail as to what you were checking in D3? IIRC that doesn't have lists or property lists as first-class objects...

Reply:
Confirmed now with @djsrv. This behavior was short-lived between 4.0 and 4.0.4

Reply:
![image](https://github.com/scummvm/scummvm/assets/365188/3e22269e-0659-4748-bb2a-d93c6b87d4c0)

Reply:
Fixed